libs/dal/src/repositories/job/job.schema.ts

```diff
@@ -143,5 +134,227 @@ jobSchema.virtual('environment', {
+ *          _subscriberId: command._subscriberId,
+ *          _templateId: command.templateId,
+ *          _environmentId: command.environmentId,
+ *          status: JobStatusEnum.COMPLETED,
+ *          type: StepTypeEnum.TRIGGER,
+ *          query['payload.' + digestKey] = DigestFilterSteps.getNestedValue(command.payload, digestKey);
+ *          updatedAt: {
+ *            $gte: this.getBackoffDate(step),
+ *          },
+ *        })
+ *    Context : alreadyHaveDigest()
+ *              type should be after _environmentId
+ *       Query : findOne({
+ *          _templateId: command.templateId,
+ *          _subscriberId: command._subscriberId,
+ *          _environmentId: command.environmentId,
+ *          type: StepTypeEnum.TRIGGER,
+ *          query['payload.' + digestKey] = DigestFilterSteps.getNestedValue(command.payload, digestKey);
+ *          updatedAt: {
+ *            $gte: this.getBackoffDate(step),
+ *          },
+ *        })
+ *
+ * Path: apps/api/src/app/events/usecases/digest-filter-steps/digest-filter-steps-regular.usecase.ts
+ *    Context : getDigest()
+ *       Query : findOne({
+ *         _templateId: command.templateId,
+ *         _subscriberId: command._subscriberId,
+ *         _environmentId: command.environmentId,
+ *         status: JobStatusEnum.DELAYED,
+ *         type: StepTypeEnum.DIGEST,
+ *          where['payload.' + digestKey] = DigestFilterSteps.getNestedValue(command.payload, digestKey);
+ *       })
+ *
+ * Path : libs/dal/src/repositories/job/job.repository.ts
+ *    Context : findJobsToDigest()
+ *       Query : find({
+ *          _templateId: templateId,
+ *          _subscriberId: subscriberId,
+ *          $or: [
+ *            { status: JobStatusEnum.COMPLETED, type: StepTypeEnum.DIGEST },
+ *            { status: JobStatusEnum.DELAYED, type: StepTypeEnum.DELAY },
+ *          ],
+ *          _environmentId: environmentId,
+ *          updatedAt: {
+ *            $gte: from,
+ *          },
+ *        })
+ *       Query : find({
+ *           updatedAt: {
+ *             $gte: from,
+ *           },
+ *           _templateId: templateId,
+ *           status: JobStatusEnum.COMPLETED,
+ *           type: StepTypeEnum.TRIGGER,
+ *           _environmentId: environmentId,
+ *           _subscriberId: subscriberId,
+ *           transactionId: {
+ *             $nin: transactionIds,
+ *           },
+ *         })
+ *       Query : update({
+ *            updatedAt: {
+ *              $gte: from,
+ *            }
+ *    Context : shouldDelayDigestJobOrMerge()
+ *       Query : find({
+ *          _subscriberId: this.convertStringToObjectId(job._subscriberId),
+ *          _templateId: job._templateId,
+ *          _environmentId: this.convertStringToObjectId(job._environmentId),
+ *          status: JobStatusEnum.DELAYED,
+ *          type: StepTypeEnum.DIGEST,
+ *          ...(digestKey && { [`payload.${digestKey}`]: digestValue }),
+ *        }
+ */
+jobSchema.index({
```

Comment:
I think this index may be "overrided" by the first query.
When you run this query, can you run https://www.mongodb.com/docs/manual/reference/method/cursor.explain/ to see which index the above query is running?